src/sdl2_compat.c

```diff
@@ -6941,6 +6941,15 @@ static SDL_InitFlags PreInitSubsystem(SDL_InitFlags flags)
         const char *old_hint;
         const char *hint;
 
+#if defined(__linux__) && defined(STEAM_RUNTIME) && STEAM_RUNTIME == 2
```

Comment:
```suggestion
#if defined(SDL2COMPAT_HAVE_X11) && defined(STEAM_RUNTIME) && STEAM_RUNTIME == 2
```

Reply:
I made the change anyway, just in case you might want to expand this from "just Steam Runtime 2" to "actually, always do this" if there are more regressions of this type than we had hoped.